### PR TITLE
[crashing] Remove Mono signal handlers when starting to handle a crash

### PR DESCRIPTION
**Pull request by alexischr**
Fixes https://github.com/mono/mono/issues/16803

### CODE REVIEW

**Comment by alexischr** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
I wonder if `pthread_atfork` should be used for this?

File: mono/mini/mini-exceptions.c [hunk 1]
```diff
@@ -3329,6 +3329,27 @@ mono_handle_native_crash (const char *signal, MonoContext *mctx, MONO_SIG_HANDLE
...
+
+	/* On some systems we get a SIGILL when calling abort (), because it might
+	 * fail to raise SIGABRT */
+	g_assert (sigaction (SIGILL, &sa, NULL) != -1);
+
+	/* Remove SIGCHLD, it uses the finalizer thread */
+	g_assert (sigaction (SIGCHLD, &sa, NULL) != -1);
+
+	/* Remove SIGQUIT, we are already dumping threads */
+	g_assert (sigaction (SIGQUIT, &sa, NULL) != -1);
```
Comment (jaykrell) [inline mono/mini/mini-exceptions.c]: `g_assert` not supposed to have side effects.

Comment (alexischr) [reply mono/mini/mini-exceptions.c]: @jaykrell This is some code that was moved and complemented, I did not want to make changes to the motif

**Comment by marek-safar** [comment]
@monojenkins backport 2019-08

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-06

**Pull request merged**